TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/fsm/StateMachine.java

```diff
@@ -1,26 +1,102 @@
 package org.firstinspires.ftc.teamcode.subsystems.fsm;
 
+import java.util.Arrays;
 import java.util.HashMap;
+import java.util.Objects;
+import org.firstinspires.ftc.teamcode.subsystems.fsm.State.Role;
+import org.firstinspires.ftc.teamcode.subsystems.fsm.State.Meta;
 
 /**
  * Represents a finite state machine.
  */
 public class StateMachine {
+    /**
+     * A runtime exception representing an insane state machine configuration encountered at runtime.
+     */
+    public static class InsaneException extends RuntimeException {
+        public InsaneException(String reason) {
+            super("Encountered an insane state machine: " + reason + ".");
+        }
+    }
+
+    /**
+     * A runtime exception representing an error that occurred during state machine construction, not
+     * runtime.
+     */
+    public static class ConstructionException extends RuntimeException {
+        public ConstructionException(Class<?> offender, String reason) {
+            super("State machine construction is invalid, caused by state '" +
+                    offender.getSimpleName() + "': " + reason + ".");
+        }
+    }
+
+    /**
+     * A runtime exception representing an unknown state.
+     * <br/>
+     * TODO: Figure out if this can be collapsed and covered under InsaneException.
+     */
+    public static class UnknownStateException extends RuntimeException {
+        public UnknownStateException(Class<?> offender) {
+            super("Valve function referenced unknown state, '" + offender.getSimpleName() + "'.");
+        }
+    }
+
     /* The flat list of states. The key is the state, the value is if it's been reached. */
-    private final HashMap<State, Boolean> states_reached = new HashMap<>();
+    private final HashMap<State, Boolean> statesReached = new HashMap<>();
```

Comment:
we want the start methods to run again- (remember the meeting milo)

Reply:
Fixed in [`b177523`](https://github.com/rh-robotics/19922-CSRC/pull/42/commits/b17752345df9febcdbc5c42d3553eb8f946ae00f).